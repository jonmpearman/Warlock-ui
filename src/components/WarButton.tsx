import React from 'react';
import styled from '@emotion/styled';

import { BaseTheme } from '../theme/default';

export const Button = ({ children, onClick }) => {
    const clickHandler = (clickEvent) => {
        const target = clickEvent?.event?.target;
        console.log(target);
        onClick?.();
    };

    return (<StyledButton onClick={clickHandler}>{children}</StyledButton>);
};

const StyledButton = styled.button`
    padding: 10px;
    background: ${BaseTheme.colors.primary.main};
    color: ${BaseTheme.colors.greyscale.white};
`;
