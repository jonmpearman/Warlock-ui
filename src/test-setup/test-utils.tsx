import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { BaseTheme } from '../theme/default';

const BaseProvider: FC = ({ children }) => {
    return (
    <ThemeProvider theme={BaseTheme}>
      {children}
    </ThemeProvider>
  )
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: BaseProvider, ...options });

export * from '@testing-library/react';
export { customRender as render };