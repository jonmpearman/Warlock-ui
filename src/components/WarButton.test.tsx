import React from 'react';
import { render } from '../test-setup/test-utils';

import { Button } from './WarButton';

test('Button Renders', () => {
    const { container } = render(<Button onClick={() => 1}>Test Button</Button>);

    expect(container).toMatchSnapshot();
});