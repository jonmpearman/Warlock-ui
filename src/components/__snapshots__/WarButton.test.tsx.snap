// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders 1`] = `
<div>
  <button
    class="css-of54i9"
  >
    Test Button
  </button>
</div>
`;
